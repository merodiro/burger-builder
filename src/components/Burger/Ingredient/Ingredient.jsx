import React, { Fragment } from 'react'
import PropTypes from 'prop-types'

import classes from './Ingredient.module.css'

const Ingredient = ({ type }) => {
  let content = null

  if (type === 'bread-top') {
    content = (
      <Fragment>
        <div className={classes.seeds1} />
        <div className={classes.seeds2} />
      </Fragment>
    )
  }
  return <div className={classes[type]}>{content}</div>
}

Ingredient.propTypes = {
  type: PropTypes.oneOf([
    'bread-bottom',
    'bread-top',
    'meat',
    'cheese',
    'salad',
    'bacon',
  ]).isRequired,
}

export default Ingredient
