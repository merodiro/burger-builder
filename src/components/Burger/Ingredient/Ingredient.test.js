import React from 'react'
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'

import Ingredient from './Ingredient'

describe('<Ingredient>', () => {
  it('renders the same output', () => {
    const types = [
      'bread-bottom',
      'bread-top',
      'meat',
      'cheese',
      'salad',
      'bacon',
    ]
    types.forEach(type => {
      const component = shallow(<Ingredient type={type} />)
      expect(component).toMatchSnapshot()
    });
  })
})
