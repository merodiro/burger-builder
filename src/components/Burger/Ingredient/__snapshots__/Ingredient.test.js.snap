// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Ingredient> renders the same output 1`] = `
<div
  className="bread-bottom"
/>
`;

exports[`<Ingredient> renders the same output 2`] = `
<div
  className="bread-top"
>
  <React.Fragment>
    <div
      className="seeds1"
    />
    <div
      className="seeds2"
    />
  </React.Fragment>
</div>
`;

exports[`<Ingredient> renders the same output 3`] = `
<div
  className="meat"
/>
`;

exports[`<Ingredient> renders the same output 4`] = `
<div
  className="cheese"
/>
`;

exports[`<Ingredient> renders the same output 5`] = `
<div
  className="salad"
/>
`;

exports[`<Ingredient> renders the same output 6`] = `
<div
  className="bacon"
/>
`;
